jerry-core/mem/mem-poolman.cpp

```diff
@@ -86,61 +165,325 @@ mem_pools_init (void)
 void
 mem_pools_finalize (void)
 {
-  JERRY_ASSERT (mem_pools == NULL);
+  mem_pools_collect_empty ();
+
+#ifndef JERRY_NDEBUG
   JERRY_ASSERT (mem_free_chunks_number == 0);
+#endif /* !JERRY_NDEBUG */
 } /* mem_pools_finalize */
 
+void
+mem_pools_collect_empty (void)
+{
+  /*
+   * Hint magic number in header of pools with free first chunks
+   */
+  const uint16_t hint_magic_num_value = 0x7e89;
```

Comment:
Ok, but certainly there has to be some decision behind it. Why not 0 for example? Is there a better number?